packages/eslint-plugin-internal/src/runtime-rules/lib-restricted-external-imports.js

```diff
@@ -0,0 +1,101 @@
+const path = require('path');
+const findUp = require('find-up');
+const minimatch = require('minimatch');
+
+/** @typedef {{ includeFiles: string, excludeFiles: string, excludeModules: string[] }} RuleSetting */
+
+/**
+ * @param {import('eslint').Rule.RuleContext} context
+ * @param {import('estree').ImportDeclaration} node
```

Comment:
@florkbr Forgot to mention, this rule doesn't check TS specific `export =` and `import = require()` syntax as documented [here](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require), since the main purpose of that syntax is interop with legacy CommonJS or AMD workflow.